Internal Error when inlining array.length expression

When I try to inline (Alt + Shift+ I) the `length` in the following code snippet
```
public static void main(String[] args) {
	String[] a = new String[0];
	int length = a.length;
	if (length > 0) {
		System.out.println("Longer"); //$NON-NLS-1$
	}
}
```
it fails with the following error:
```
java.lang.reflect.InvocationTargetException
	at org.eclipse.jface.operation.ModalContext.run(ModalContext.java:396)
	at org.eclipse.jface.dialogs.ProgressMonitorDialog.run(ProgressMonitorDialog.java:470)
	at org.eclipse.ui.internal.progress.ProgressMonitorJobsDialog.run(ProgressMonitorJobsDialog.java:230)
	at org.eclipse.ui.internal.progress.ProgressManager.lambda$26(ProgressManager.java:819)
	at org.eclipse.swt.custom.BusyIndicator.showWhile(BusyIndicator.java:67)
	at org.eclipse.ui.internal.progress.ProgressManager.busyCursorWhile(ProgressManager.java:849)
	at org.eclipse.ui.internal.progress.ProgressManager.busyCursorWhile(ProgressManager.java:828)
	at org.eclipse.ltk.ui.refactoring.RefactoringWizardOpenOperation.checkInitialConditions(RefactoringWizardOpenOperation.java:222)
	at org.eclipse.ltk.ui.refactoring.RefactoringWizardOpenOperation.lambda$0(RefactoringWizardOpenOperation.java:173)
	at org.eclipse.swt.custom.BusyIndicator.showWhile(BusyIndicator.java:67)
	at org.eclipse.ltk.ui.refactoring.RefactoringWizardOpenOperation.run(RefactoringWizardOpenOperation.java:209)
	at org.eclipse.ltk.ui.refactoring.RefactoringWizardOpenOperation.run(RefactoringWizardOpenOperation.java:126)
	at org.eclipse.jdt.internal.ui.refactoring.actions.RefactoringStarter.activate(RefactoringStarter.java:41)
	at org.eclipse.jdt.internal.corext.refactoring.RefactoringExecutionStarter.startInlineTempRefactoring(RefactoringExecutionStarter.java:351)
	at org.eclipse.jdt.ui.actions.InlineTempAction.tryInlineTemp(InlineTempAction.java:121)
	at org.eclipse.jdt.ui.actions.InlineAction.run(InlineAction.java:120)
	at org.eclipse.jdt.ui.actions.SelectionDispatchAction.dispatchRun(SelectionDispatchAction.java:278)
	at org.eclipse.jdt.ui.actions.SelectionDispatchAction.run(SelectionDispatchAction.java:252)
	at org.eclipse.jface.action.Action.runWithEvent(Action.java:474)
	at org.eclipse.jface.commands.ActionHandler.execute(ActionHandler.java:124)
	at org.eclipse.ui.internal.handlers.E4HandlerProxy.execute(E4HandlerProxy.java:98)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.eclipse.e4.core.internal.di.MethodRequestor.execute(MethodRequestor.java:56)
	at org.eclipse.e4.core.internal.di.InjectorImpl.invokeUsingClass(InjectorImpl.java:299)
	at org.eclipse.e4.core.internal.di.InjectorImpl.invoke(InjectorImpl.java:233)
	at org.eclipse.e4.core.contexts.ContextInjectionFactory.invoke(ContextInjectionFactory.java:174)
	at org.eclipse.e4.core.commands.internal.HandlerServiceHandler.execute(HandlerServiceHandler.java:165)
	at org.eclipse.core.commands.Command.executeWithChecks(Command.java:488)
	at org.eclipse.core.commands.ParameterizedCommand.executeWithChecks(ParameterizedCommand.java:485)
	at org.eclipse.e4.core.commands.internal.HandlerServiceImpl.executeHandler(HandlerServiceImpl.java:204)
	at org.eclipse.e4.ui.bindings.keys.KeyBindingDispatcher.executeCommand(KeyBindingDispatcher.java:308)
	at org.eclipse.e4.ui.bindings.keys.KeyBindingDispatcher.press(KeyBindingDispatcher.java:569)
	at org.eclipse.e4.ui.bindings.keys.KeyBindingDispatcher.processKeyEvent(KeyBindingDispatcher.java:644)
	at org.eclipse.e4.ui.bindings.keys.KeyBindingDispatcher.filterKeySequenceBindings(KeyBindingDispatcher.java:439)
	at org.eclipse.e4.ui.bindings.keys.KeyBindingDispatcher$KeyDownFilter.handleEvent(KeyBindingDispatcher.java:96)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:91)
	at org.eclipse.swt.widgets.Display.filterEvent(Display.java:1319)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1173)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1198)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1183)
	at org.eclipse.swt.widgets.Control.traverse(Control.java:4234)
	at org.eclipse.swt.widgets.Control.translateMnemonic(Control.java:4084)
	at org.eclipse.swt.widgets.Composite.translateMnemonic(Composite.java:1259)
	at org.eclipse.swt.widgets.Control.translateMnemonic(Control.java:4102)
	at org.eclipse.swt.widgets.Display.translateMnemonic(Display.java:4991)
	at org.eclipse.swt.widgets.Display.filterMessage(Display.java:1337)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3708)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1151)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:339)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1042)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:153)
	at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:639)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:339)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:546)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:173)
	at org.eclipse.ui.internal.ide.application.IDEApplication.start(IDEApplication.java:152)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:208)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:143)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:109)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:439)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:271)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:668)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:605)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1481)
Caused by: java.lang.IllegalArgumentException: Invalid identifier : >String[]<
	at org.eclipse.jdt.core.dom.SimpleName.setIdentifier(SimpleName.java:247)
	at org.eclipse.jdt.core.dom.AST.newSimpleName(AST.java:2935)
	at org.eclipse.jdt.core.dom.AST.newName(AST.java:2561)
	at org.eclipse.jdt.internal.corext.refactoring.code.InlineTempRefactoring.createFullyQualifiedName(InlineTempRefactoring.java:779)
	at org.eclipse.jdt.internal.corext.refactoring.code.InlineTempRefactoring.createFullyQualifiedName(InlineTempRefactoring.java:752)
	at org.eclipse.jdt.internal.corext.refactoring.code.InlineTempRefactoring.getAlternativeQualifications(InlineTempRefactoring.java:658)
	at org.eclipse.jdt.internal.corext.refactoring.code.InlineTempRefactoring.checkClashes(InlineTempRefactoring.java:264)
	at org.eclipse.jdt.internal.corext.refactoring.code.InlineTempRefactoring.checkInitialConditions(InlineTempRefactoring.java:250)
	at org.eclipse.ltk.core.refactoring.CheckConditionsOperation.run(CheckConditionsOperation.java:84)
	at org.eclipse.core.internal.resources.Workspace.run(Workspace.java:2457)
	at org.eclipse.core.internal.resources.Workspace.run(Workspace.java:2482)
	at org.eclipse.ltk.internal.ui.refactoring.WorkbenchRunnableAdapter.run(WorkbenchRunnableAdapter.java:89)
	at org.eclipse.jface.operation.ModalContext$ModalContextThread.run(ModalContext.java:124)
```

--- a/org.eclipse.jdt.core.manipulation/core extension/org/eclipse/jdt/internal/corext/refactoring/code/InlineTempRefactoring.java
+++ b/org.eclipse.jdt.core.manipulation/core extension/org/eclipse/jdt/internal/corext/refactoring/code/InlineTempRefactoring.java
@@ -262,16 +262,18 @@ public class InlineTempRefactoring extends Refactoring {
 			for (Name name : initializerNames) {
 				if (clashesWithNewVariables(newVariables, name)) {
 					List<Expression> alternativeQualifications= getAlternativeQualifications(reference, name);
-					boolean inlinable= false;
-					for (Expression alternative : alternativeQualifications) {
-						if (!clashesWithNewVariables(newVariables, alternative)) {
-							inlinable= true;
-							break;
+					if (alternativeQualifications.size() > 0) {
+						boolean inlinable= false;
+						for (Expression alternative : alternativeQualifications) {
+							if (!clashesWithNewVariables(newVariables, alternative)) {
+								inlinable= true;
+								break;
+							}
 						}
-					}
-					if (!inlinable) {
-						return RefactoringStatus.createFatalErrorStatus(
-								Messages.format(RefactoringCoreMessages.InlineTemRefactoring_error_message_inliningClashes, name));
+						if (!inlinable) {
+							return RefactoringStatus.createFatalErrorStatus(
+									Messages.format(RefactoringCoreMessages.InlineTemRefactoring_error_message_inliningClashes, name));
+						}
 					}
 				}
 			}
@@ -652,13 +654,19 @@ public class InlineTempRefactoring extends Refactoring {
 			}
 		} else if (initializerName instanceof QualifiedName){
 			QualifiedName initializerQualifiedName = (QualifiedName) initializerName;
-			SimpleName simpleName= initializerQualifiedName.getName();
-			int i= findCommonDeclaringClassIndex(reference, initializerQualifiedName);
-			ITypeBinding initializerQualifiedNameTypeBinding= initializerQualifiedName.getQualifier().resolveTypeBinding();
-			ans.add(createFullyQualifiedName(simpleName, initializerQualifiedNameTypeBinding, false));
-			ans.add(createFullyQualifiedName(simpleName, initializerQualifiedNameTypeBinding, i, false));
-			ans.add(createFullyQualifiedName(simpleName, initializerQualifiedNameTypeBinding, true));
-			ans.add(createFullyQualifiedName(simpleName, initializerQualifiedNameTypeBinding, i, true));
+			IBinding resolveBinding= initializerQualifiedName.resolveBinding();
+			if (resolveBinding instanceof IVariableBinding resolvedVariableBinding) {
+				boolean isStatic= Modifier.isStatic(resolvedVariableBinding.getModifiers());
+				if (isStatic) {
+					SimpleName simpleName= initializerQualifiedName.getName();
+					int i= findCommonDeclaringClassIndex(reference, initializerQualifiedName);
+					ITypeBinding initializerQualifiedNameTypeBinding= initializerQualifiedName.getQualifier().resolveTypeBinding();
+					ans.add(createFullyQualifiedName(simpleName, initializerQualifiedNameTypeBinding, false));
+					ans.add(createFullyQualifiedName(simpleName, initializerQualifiedNameTypeBinding, i, false));
+					ans.add(createFullyQualifiedName(simpleName, initializerQualifiedNameTypeBinding, true));
+					ans.add(createFullyQualifiedName(simpleName, initializerQualifiedNameTypeBinding, i, true));
+				}
+			}
 		}
 		return ans;
 	}

--- a/org.eclipse.jdt.ui.tests.refactoring/resources/InlineTemp/canInline/A_test65_in.java
+++ b/org.eclipse.jdt.ui.tests.refactoring/resources/InlineTemp/canInline/A_test65_in.java
@@ -1,0 +1,11 @@
+package p;
+
+public class A {
+	public static void main(String[] args) {
+		String[] a= new String[0];
+		int length= a.length;
+		if (length > 0) {
+			System.out.println("Longer"); //$NON-NLS-1$
+		}
+	}
+}

--- a/org.eclipse.jdt.ui.tests.refactoring/resources/InlineTemp/canInline/A_test65_out.java
+++ b/org.eclipse.jdt.ui.tests.refactoring/resources/InlineTemp/canInline/A_test65_out.java
@@ -1,0 +1,10 @@
+package p;
+
+public class A {
+	public static void main(String[] args) {
+		String[] a= new String[0];
+		if (a.length > 0) {
+			System.out.println("Longer"); //$NON-NLS-1$
+		}
+	}
+}

--- a/org.eclipse.jdt.ui.tests.refactoring/test cases/org/eclipse/jdt/ui/tests/refactoring/InlineTempTests.java
+++ b/org.eclipse.jdt.ui.tests.refactoring/test cases/org/eclipse/jdt/ui/tests/refactoring/InlineTempTests.java
@@ -347,66 +347,6 @@ public class InlineTempTests extends GenericRefactoringTest {
 		helper1(5, 43, 5, 46);
 	}
 
-	public void test62() throws Exception {
-		//https://bugs.eclipse.org/bugs/show_bug.cgi?id=367536
-		helper1(14, 13, 14, 14);
-	}
-
-	public void test63() throws Exception {
-		//https://bugs.eclipse.org/bugs/show_bug.cgi?id=367536
-		helper1(11, 13, 11, 14);
-	}
-
-	public void test64() throws Exception {
-		//https://bugs.eclipse.org/bugs/show_bug.cgi?id=367536
-		helper1(7, 13, 7, 14);
-	}
-
-	public void test53() throws Exception {
-		//https://bugs.eclipse.org/bugs/show_bug.cgi?id=367536
-		helper1(12, 13, 12, 14);
-	}
-
-	public void test54() throws Exception {
-		//https://bugs.eclipse.org/bugs/show_bug.cgi?id=367536
-		helper1(7, 13, 7, 14);
-	}
-
-	public void test55() throws Exception {
-		//https://bugs.eclipse.org/bugs/show_bug.cgi?id=367536
-		helper1(11, 13, 11, 14);
-	}
-
-	public void test56() throws Exception {
-		//https://bugs.eclipse.org/bugs/show_bug.cgi?id=367536
-		helper1(12, 13, 12, 14);
-	}
-
-	public void test57() throws Exception {
-		//https://bugs.eclipse.org/bugs/show_bug.cgi?id=367536
-		helper1(9, 26, 9, 27);
-	}
-
-	public void test58() throws Exception {
-		//https://bugs.eclipse.org/bugs/show_bug.cgi?id=367536
-		helper1(9, 26, 9, 27);
-	}
-
-	public void test59() throws Exception {
-		//https://bugs.eclipse.org/bugs/show_bug.cgi?id=367536
-		helper1(11, 11, 11, 12);
-	}
-
-	public void test60() throws Exception {
-		//https://bugs.eclipse.org/bugs/show_bug.cgi?id=367536
-		helper1(7, 13, 7, 14);
-	}
-
-	public void test61() throws Exception {
-		//https://bugs.eclipse.org/bugs/show_bug.cgi?id=367536
-		helper1(9, 13, 9, 14);
-	}
-
 	@Test
 	public void test41() throws Exception {
 		// https://bugs.eclipse.org/bugs/show_bug.cgi?id=335173
@@ -477,6 +417,84 @@ public class InlineTempTests extends GenericRefactoringTest {
 	public void test52() throws Exception {
 		//https://bugs.eclipse.org/bugs/show_bug.cgi?id=434747
 		helper1(4, 14, 4, 15);
+	}
+
+	@Test
+	public void test53() throws Exception {
+		//https://bugs.eclipse.org/bugs/show_bug.cgi?id=367536
+		helper1(12, 13, 12, 14);
+	}
+
+	@Test
+	public void test54() throws Exception {
+		//https://bugs.eclipse.org/bugs/show_bug.cgi?id=367536
+		helper1(7, 13, 7, 14);
+	}
+
+	@Test
+	public void test55() throws Exception {
+		//https://bugs.eclipse.org/bugs/show_bug.cgi?id=367536
+		helper1(11, 13, 11, 14);
+	}
+
+	@Test
+	public void test56() throws Exception {
+		//https://bugs.eclipse.org/bugs/show_bug.cgi?id=367536
+		helper1(12, 13, 12, 14);
+	}
+
+	@Test
+	public void test57() throws Exception {
+		//https://bugs.eclipse.org/bugs/show_bug.cgi?id=367536
+		helper1(9, 26, 9, 27);
+	}
+
+	@Test
+	public void test58() throws Exception {
+		//https://bugs.eclipse.org/bugs/show_bug.cgi?id=367536
+		helper1(9, 26, 9, 27);
+	}
+
+	@Test
+	public void test59() throws Exception {
+		//https://bugs.eclipse.org/bugs/show_bug.cgi?id=367536
+		helper1(11, 11, 11, 12);
+	}
+
+	@Test
+	public void test60() throws Exception {
+		//https://bugs.eclipse.org/bugs/show_bug.cgi?id=367536
+		helper1(7, 13, 7, 14);
+	}
+
+	@Test
+	public void test61() throws Exception {
+		//https://bugs.eclipse.org/bugs/show_bug.cgi?id=367536
+		helper1(9, 13, 9, 14);
+	}
+
+	@Test
+	public void test62() throws Exception {
+		//https://bugs.eclipse.org/bugs/show_bug.cgi?id=367536
+		helper1(14, 13, 14, 14);
+	}
+
+	@Test
+	public void test63() throws Exception {
+		//https://bugs.eclipse.org/bugs/show_bug.cgi?id=367536
+		helper1(11, 13, 11, 14);
+	}
+
+	@Test
+	public void test64() throws Exception {
+		//https://bugs.eclipse.org/bugs/show_bug.cgi?id=367536
+		helper1(7, 13, 7, 14);
+	}
+
+	@Test
+	public void test65() throws Exception {
+		//https://github.com/eclipse-jdt/eclipse.jdt.ui/issues/1705
+		helper1(6, 13, 6, 19);
 	}
 
 	//------

